docs/usage/customsessions.md

```diff
@@ -0,0 +1,145 @@
+  private client: redis.RedisClient;
+  private getAsync;
+
+  constructor() {
+    // Create a new redis client
+    this.client = redis.createClient();
+    // Use Node's `promisify` to have redis return a promise from the `get` method
+    this.getAsync = promisify(this.client.get).bind(this.client);
+  }
+
+  /*
+    The storeCallback takes in the Session, and sets a stringified version of it on the redis store
+    If the session can be stored, we return true
+    Otherwise, return false
+  */
+  storeCallback = async (session: Session) => {
+    try {
+      if (this.client.set(session.id, JSON.stringify(session))) {
+        return true;
+      } else {
+        return false
+      }
+    } catch (err) {
+      // throw errors, and handle them gracefully in your application
+      throw new Error(err)
+    }
+  };
+
+  /*
+    The loadCallback takes in the id, and uses the getAsync method to access the session data
+     If a stored session exists, it's parsed and returned
+     Otherwise, we return undefined
+  */
+  loadCallback = async (id: string) => {
+    try {
+      let reply = await this.getAsync(id);
+      if (reply) {
+        return JSON.parse(reply);
+      } else {
+        return undefined
+      }
+    } catch (err) {
+      throw new Error(err)
+    }
+  };
+
+  /*
+    The deleteCallback takes in the id, and uses the redis `del` method to delete it from the store
+    If the session can be deleted, we return true
+    Otherwise, we return false
+  */
+  deleteCallback = async (id: string) => {
+    try {
+      if (this.client.del(id)) {
+        return true;
+      } else {
+        return false
+      }
+    } catch (err) {
+      throw new Error(err)
+    }
+  };
+}
+
+// Export the class
+export default RedisStore;
+```
+
+### Create a new `CustomSessionStorage` to use in `Shopify.Context`
+
+Now we can import our custom storage class into our `index.ts` and use it to setup our storage solution in `Shopify.Context`.
+
+```ts
+/* index.ts */
+
+import Shopify, {ApiVersion, AuthQuery} from '@shopify/shopify-api/dist';
+import {CustomSessionStorage} from '@shopify/shopify-api/dist/auth/session';
+// Import our custom storage class
+import RedisStore from './redis-store';
+
+require('dotenv').config();
+
+// Create a new instance of the custom storage class
+const sessionStorage = new RedisStore();
+
+const {API_KEY, API_SECRET_KEY, SCOPES, SHOP, HOST} = process.env;
+
+// Setup Shopify.Context with CustomSessionStorage
+Shopify.Context.initialize({
+  API_KEY,
+  API_SECRET_KEY,
+  SCOPES,
+  HOST_NAME: HOST,
+  IS_EMBEDDED_APP: true,
+  API_VERSION: ApiVersion.unstable,
+  // Pass the sessionStorage methods to pass into a new instance of `CustomSessionStorage`
+  SESSION_STORAGE: new CustomSessionStorage(
+    sessionStorage.storeCallback,
+    sessionStorage.loadCallback,
+    sessionStorage.deleteCallback,
+  ),
+});
+
+// Your app will now use your new storage solution to access and manage Sessions
+
+```
```

Comment:
```ts
-import Shopify, {ApiVersion, AuthQuery} from '@shopify/shopify-api/dist';
-import {CustomSessionStorage} from '@shopify/shopify-api/dist/auth/session';
-  SESSION_STORAGE: new CustomSessionStorage(
```
This may be better.
```ts
+import Shopify, {ApiVersion} from '@shopify/shopify-api';
+  SESSION_STORAGE: new Shopify.Session.CustomSessionStorage(
```